### ISSUE TITLE
mono version crashes on Mac OSX when the Commit dialog is opened

### ISSUE THREAD

**Issue by zpangwin** [gitextensions/gitextensions#7578]
Have used GitExtensions extensively on Windows. Have also used it quite a bit under mono on Linux Mint without any significant issues. On Mac OSX, I'm attempting to follow the instructions [here](https://git-extensions-documentation.readthedocs.io/en/release-2.51/getting_started.html#installation-macos-2-5x-only); I am able to launch the Browse dialog but clicking Commit toolbar button or launching from CLI with the "commit" argument, I see the Commit dialog open briefly and then the entire GitExtensions/mono process crashes.

I realize that v2.50 may no longer be officially supported. I would be more than happy with any unofficial guesses / suggestions / comparing notes with anyone who has a working setup on mac. This might also be nice for anyone coming in from google etc after encountering the same issue.

Git Extensions version: [v2.51.05](https://github.com/gitextensions/gitextensions/releases/tag/v2.51.05) (also tried with 2.48.05). I am using the *-Mono.zip for both.
System: macOS Mojave 10.14.6 / macbook pro
Mono version: 6.6.0.161 (see below for exact terminal output)

```
$ mono --version
Mono JIT compiler version 6.6.0.161 (2019-08/bef1e633581 Tue Dec  3 13:05:34 EST 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	Interpreter:   yes
	LLVM:          yes(610)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
```

Additionally, during setup I had to apply the following steps to get the main Browse dialog to work. Didn't test everything but the New Branch and Merge Branch dialogs seemed to work ok after these steps; just not Commit.
1. After extracting the *Mono.zip archive somewhere under user folder (e.g. $USER/Bin/GitExtensions), I added to the PATH variable and exported path in ~/.bash_profile
2. Created backup: `cp -a gitext.sh gitext.sh.orig`
3. Updated gitext.sh script to force 32-bit version of mono per Mac instructions above: `mono --arch=32 "$DIR/GitExtensions.exe" "$@"   &`
4. Under the ./GitExtensions/Plugins folder, I had to rename one of the dll's to prevent GitExt Browse dialog from hanging on load: `mv TeamCityIntegration.dll TeamCityIntegration.dll.bak`
5. Fixed perms under GitExtensions dir:

```
# note I am also using the coreutils (aka GNU) versions of find/chmod/etc and have
# those defined in my exported PATH as well.
# so if the params are different than BSD/Apple versions then that's why
find . -type f -exec chmod a+r "{}" \;
find . -type d -exec chmod a+rx "{}" \;
find . -type f -iname "*.dll" -o -iname "*.exe" -o -iname "*.sh" -exec chmod a+rx "{}" \;
```

Here is what I get if I try to launch the Commit dialog (showing from CLI but same error when launching from toolbar button on the Browse dialog):

```
$ gitext.sh commit
# <note: dialog appears briefly at this point then immediately closes and disappears>

$ 2019-12-30 10:49:08.576 mono32[21955:146974] *** Assertion failure in +[NSServiceViewController currentAppIsViewService], /BuildRoot/Library/Caches/com.apple.xbs/Sources/ViewBridge/ViewBridge-401.1/NSViewService.m:131
2019-12-30 10:49:08.576 mono32[21955:146974] *** Assertion failure in +[NSServiceViewController currentAppIsViewService], /BuildRoot/Library/Caches/com.apple.xbs/Sources/ViewBridge/ViewBridge-401.1/NSViewService.m:131
System.ArgumentException: Rectangle '{X=0,Y=274,Width=397,Height=0}' cannot have a width or height equal to 0.
  at System.Drawing.Drawing2D.LinearGradientBrush..ctor (System.Drawing.Rectangle rect, System.Drawing.Color color1, System.Drawing.Color color2, System.Drawing.Drawing2D.LinearGradientMode linearGradientMode) [0x00050] in <73e252f4c0b44602b173b66913b8ad0e>:0
  at (wrapper remoting-invoke-with-check) System.Drawing.Drawing2D.LinearGradientBrush..ctor(System.Drawing.Rectangle,System.Drawing.Color,System.Drawing.Color,System.Drawing.Drawing2D.LinearGradientMode)
  at System.Windows.Forms.ToolStripProfessionalRenderer.OnRenderToolStripPanelBackground (System.Windows.Forms.ToolStripPanelRenderEventArgs e) [0x0002c] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.ToolStripRenderer.DrawToolStripPanelBackground (System.Windows.Forms.ToolStripPanelRenderEventArgs e) [0x00000] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.ToolStripPanel.OnPaintBackground (System.Windows.Forms.PaintEventArgs e) [0x00019] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.Control.WmPaint (System.Windows.Forms.Message& m) [0x0005f] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.Control.WndProc (System.Windows.Forms.Message& m) [0x001a4] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.ScrollableControl.WndProc (System.Windows.Forms.Message& m) [0x00000] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.ContainerControl.WndProc (System.Windows.Forms.Message& m) [0x00027] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.Control+ControlWindowTarget.OnMessage (System.Windows.Forms.Message& m) [0x00000] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.Control+ControlNativeWindow.WndProc (System.Windows.Forms.Message& m) [0x0000b] in <41691a4731114b7d9842444885fc088e>:0
  at System.Windows.Forms.NativeWindow.WndProc (System.IntPtr hWnd, System.Windows.Forms.Msg msg, System.IntPtr wParam, System.IntPtr lParam) [0x00085] in <41691a4731114b7d9842444885fc088e>:0

Unhandled Exception:
System.ObjectDisposedException: The CancellationTokenSource has been disposed.
  at System.Threading.CancellationTokenSource.Cancel (System.Boolean throwOnFirstException) [0x00000] in <cc0e10f6687f4e3da2c8dd4cda700ae3>:0
  at System.Threading.CancellationTokenSource.Cancel () [0x00000] in <cc0e10f6687f4e3da2c8dd4cda700ae3>:0
  at GitCommands.AsyncLoader.Cancel () [0x00008] in <3eb8e9d083d04ef3adf987a8a1b67c27>:0
  at GitUI.CommandsDialogs.FormCommit.Dispose (System.Boolean disposing) [0x00003] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at System.ComponentModel.Component.Dispose () [0x00000] in <d318895abf694c2b9d1de16eb38d26da>:0
  at GitUI.GitUICommands+<>c__DisplayClass332_0.<StartCommitDialog>b__0 () [0x00034] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.DoActionOnRepo (System.Windows.Forms.IWin32Window owner, System.Boolean requiresValidWorkingDir, System.Boolean changesRepo, GitUIPluginInterfaces.GitUIEventHandler preEvent, GitUIPluginInterfaces.GitUIPostActionEventHandler postEvent, System.Func`1[TResult] action) [0x0002d] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.StartCommitDialog (System.Windows.Forms.IWin32Window owner, System.Boolean showOnlyWhenChanges) [0x00028] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.StartCommitDialog (System.Boolean showOnlyWhenChanges) [0x00000] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.Commit (System.Collections.Generic.Dictionary`2[TKey,TValue] arguments) [0x0000c] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.RunCommandBasedOnArgument (System.String[] args, System.Collections.Generic.Dictionary`2[TKey,TValue] arguments) [0x0065c] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.RunCommand (System.String[] args) [0x000c6] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitExtensions.Program.Main () [0x001ea] in <e36e0ba6ef3a4658bf85324e88a3bf22>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.ObjectDisposedException: The CancellationTokenSource has been disposed.
  at System.Threading.CancellationTokenSource.Cancel (System.Boolean throwOnFirstException) [0x00000] in <cc0e10f6687f4e3da2c8dd4cda700ae3>:0
  at System.Threading.CancellationTokenSource.Cancel () [0x00000] in <cc0e10f6687f4e3da2c8dd4cda700ae3>:0
  at GitCommands.AsyncLoader.Cancel () [0x00008] in <3eb8e9d083d04ef3adf987a8a1b67c27>:0
  at GitUI.CommandsDialogs.FormCommit.Dispose (System.Boolean disposing) [0x00003] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at System.ComponentModel.Component.Dispose () [0x00000] in <d318895abf694c2b9d1de16eb38d26da>:0
  at GitUI.GitUICommands+<>c__DisplayClass332_0.<StartCommitDialog>b__0 () [0x00034] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.DoActionOnRepo (System.Windows.Forms.IWin32Window owner, System.Boolean requiresValidWorkingDir, System.Boolean changesRepo, GitUIPluginInterfaces.GitUIEventHandler preEvent, GitUIPluginInterfaces.GitUIPostActionEventHandler postEvent, System.Func`1[TResult] action) [0x0002d] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.StartCommitDialog (System.Windows.Forms.IWin32Window owner, System.Boolean showOnlyWhenChanges) [0x00028] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.StartCommitDialog (System.Boolean showOnlyWhenChanges) [0x00000] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.Commit (System.Collections.Generic.Dictionary`2[TKey,TValue] arguments) [0x0000c] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.RunCommandBasedOnArgument (System.String[] args, System.Collections.Generic.Dictionary`2[TKey,TValue] arguments) [0x0065c] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitUI.GitUICommands.RunCommand (System.String[] args) [0x000c6] in <4ca0a9d3b3f3459fb67691431a63cda3>:0
  at GitExtensions.Program.Main () [0x001ea] in <e36e0ba6ef3a4658bf85324e88a3bf22>:0
```

**Comment by zpangwin** [comment]
Also, I tried going into my ~/.config/GitExtensions/GitExtensions folder and deleting `GitExtensions.settings*` and then relaunching -- under the premise that this might reset window positions to their defaults in case if I inadvertently messed something up somehow. 

Unfortunately, this had no effect whatsoever.

**Comment by gerhardol** [comment]
> I realize that v2.50 may no longer be officially supported

If there is a need, there will be new releases.
The problem is that no one in the core team is using the Mono version, so it is not actively developed.
A few hints here too:
https://github.com/gitextensions/gitextensions/wiki/How-To%3A-run-Git-Extensions-on-Linux
The doc should be updated.

The crash is in Mono, as this seem to work in other .NET runtime, the normal explanation is Mono issues.
Any possibility to test with other Mono versions?

**Comment by zpangwin** [comment]
> If there is a need, there will be new releases.

Good to know; I hadn't realized that this was still a possibility for the mono-based version / v2.x line.

> The crash is in Mono, as this seem to work in other .NET runtime, the normal explanation is Mono issues.

Makes sense. I will also open this issue against mono project then. If there are no objections, I would prefer to still leave this one open for the time being until I've had a chance to test and see if a workaround might be possible.

> Any possibility to test with other Mono versions?

Sure, I can definitely try that. Is there any particular version(s) of Mono / Git Extensions that you would recommend I test with first? The doc you linked to just says to use "at least in 5.4" (of Mono) under the Mac section; I was using the latest stable version for Mac but I can try older and/or less stable versions. Otherwise, my thinking for now is to try with the Preview and nightly builds first. Then if those don't pan out, I'll try to check what version combo I had working at home under Linux Mint 19 for Mono / GitExt and see if I can at least confirm the issue is isolated to Macs.

**Comment by RussKie** [comment]
We will take Mono-related changes into the 2.5x branch, though there won't be a parity with the Windows version.

Happy to keep the issue open too.

Last time I used Ubuntu (I think it was 18) I had run GE on Mono 5.14, which was fresh from the press back then.

**Comment by gerhardol** [comment]
> any particular version(s) of Mono / Git Extensions that you would recommend I test with first? 

No info myself, but it seems like 5.x versions have been working for other users.

**Comment by zpangwin** [comment]
> We will take Mono-related changes into the 2.5x branch, though there won't be a parity with the Windows version.
> 
> Happy to keep the issue open too.
> 
> Last time I used Ubuntu (I think it was 18) I had run GE on Mono 5.14, which was fresh from the press back then.

Cool thanks. And yeah, don't recall ever seeing this issue on Ubuntu/Mint. On those boxes, I have been using GE for well over a year. I think I have Mono setup to come from a PPA but don't remember if its stable/preview/etc... have used both GE 2.48.x and 2.50.x on nix and my biggest "issue" that I remember was that I had to rename a dll during install (same as with mac).

**Comment by zpangwin** [comment]
Also, I went ahead and opened [Mono issue #18330](https://github.com/mono/mono/issues/18330) to track this on that side.

**Comment by zpangwin** [comment]
Confirmed same behavior under Mono Preview v6.8.0.85

```
$ mono --version
Mono JIT compiler version 6.8.0.85 (2019-10/99e3aed6ec3 Wed Jan  1 20:25:30 EST 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	Interpreter:   yes
	LLVM:          yes(610)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
```

**Comment by zpangwin** [comment]
Same thing with nightly build MonoFramework-MDK-master@6f248326465-dirty-6.11.0.145.macos10.xamarin.universal.pkg  \| 2020-01-02 08:19 \| 338M from [here](https://download.mono-project.com/archive/nightly/macos-10-universal/?C=M;O=D)

```
mono --version
Mono JIT compiler version 6.11.0.145 (master/6f248326465 Mon Dec 30 23:16:00 EST 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	Interpreter:   yes
	LLVM:          yes(610)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
```

**Comment by zpangwin** [comment]
The original instruction set that I was referencing [here](https://git-extensions-documentation.readthedocs.io/en/release-2.51/getting_started.html#installation-macos-2-5x-only) mentioned version 4.6.1 somewhere and I tried that. This version was also from prior to the mono32 / mono64 split. Unfortunately, it generates the same stacktrace as the current stable/preview/nightly versions.

```
$ mono --version
Mono JIT compiler version 4.6.1 (mono-4.6.0-branch-c8sr0/ef43c15 Mon Oct  3 14:35:32 EDT 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  x86
	Disabled:      none
	Misc:          softdebug
	LLVM:          yes(3.6.0svn-mono-master/8b1520c)
	GC:            sgen
```

RussKie also mentioned that he was using Mono v5.14 on Ubuntu so tried that too:

```
$ mono --version
Mono JIT compiler version 5.14.0.177 (2018-04/f3a2216b65a Fri Aug  3 09:28:16 EDT 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-master/8b1520c8aae)
	GC:            sgen (concurrent by default)
``` 

Unfortunately, I get the same stacktrace yet again.

I have also attempted this on various different repos including a dummy one created under /tmp so I don't think it is anything to do with a specific git repo. but just in case it matters, I have this for my git version (not the latest but released 2019-02-24 so reasonably non-ancient at least):

```
$ git --version
git version 2.21.0 (Apple Git-122.2)
```

**Comment by zpangwin** [comment]
Also confirmed that the commit dialog on my Linux Mint 19.3 boxes works just fine.

* One box had: GitExt v2.51.05 + Mono v6.6.0.161
* The other had: GitExt v2.51.05 + Mono v5.10.1.20

this is from the box with mono v6.6

```
# On Linux Mint (for comparison):

$ cat /etc/apt/sources.list.d/mono-official-stable.list 
deb https://download.mono-project.com/repo/ubuntu stable-bionic main
```

and it looks like I had installed these packages: `mono-runtime mono-devel mono-complete referenceassemblies-pcl ca-certificates-mono mono-xsp4`

**Comment by t00** [comment]
I can confirm the occasional crash on Linux (Manjaro). The issue comes and goes (had it in the past), possibly related to installed fonts and/or mono version.

Currently crashes on Mono JIT compiler version 6.12.0

The crash log is as follows:

```
System.Exception: Generic Error [GDI+ status: GenericError]
  at System.Drawing.GDIPlus.CheckStatus (System.Drawing.Status status) [0x00079] in <9c1c445ab98849e19c6ec36da81bc887>:0 
  at System.Drawing.Graphics.FillRectangle (System.Drawing.Brush brush, System.Single x, System.Single y, System.Single width, System.Single height) [0x00025] in <9c1c445ab98849e19c6ec36da81bc887>:0 
  at System.Drawing.Graphics.FillRectangle (System.Drawing.Brush brush, System.Drawing.RectangleF rect) [0x0002c] in <9c1c445ab98849e19c6ec36da81bc887>:0 
  at (wrapper remoting-invoke-with-check) System.Drawing.Graphics.FillRectangle(System.Drawing.Brush,System.Drawing.RectangleF)
  at ICSharpCode.TextEditor.TextView.DrawDocumentWord (System.Drawing.Graphics g, System.String word, System.Drawing.Point position, System.Drawing.Font font, System.Drawing.Color foreColor, System.Drawing.Brush backBrush) [0x000c0] in <28e4e294b36d4d89be777f567b58b05c>:0 
  at ICSharpCode.TextEditor.TextView.PaintLinePart (System.Drawing.Graphics g, System.Int32 lineNumber, System.Int32 startColumn, System.Int32 endColumn, System.Drawing.Rectangle lineRectangle, System.Int32 physicalXPos) [0x0056c] in <28e4e294b36d4d89be777f567b58b05c>:0 
  at ICSharpCode.TextEditor.TextView.PaintDocumentLine (System.Drawing.Graphics g, System.Int32 lineNumber, System.Drawing.Rectangle lineRectangle) [0x000eb] in <28e4e294b36d4d89be777f567b58b05c>:0 
  at ICSharpCode.TextEditor.TextView.Paint (System.Drawing.Graphics g, System.Drawing.Rectangle rect) [0x00102] in <28e4e294b36d4d89be777f567b58b05c>:0 
  at ICSharpCode.TextEditor.TextArea.OnPaint (System.Windows.Forms.PaintEventArgs e) [0x001ce] in <28e4e294b36d4d89be777f567b58b05c>:0 
  at System.Windows.Forms.Control.WmPaint (System.Windows.Forms.Message& m) [0x0007c] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control.WndProc (System.Windows.Forms.Message& m) [0x001a4] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control+ControlWindowTarget.OnMessage (System.Windows.Forms.Message& m) [0x00000] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control+ControlNativeWindow.WndProc (System.Windows.Forms.Message& m) [0x0000b] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.NativeWindow.WndProc (System.IntPtr hWnd, System.Windows.Forms.Msg msg, System.IntPtr wParam, System.IntPtr lParam) [0x00085] in <4b6c441381804088ab0fff508a3fbabf>:0 
System.ObjectDisposedException: The CancellationTokenSource has been disposed.
  at System.Threading.CancellationTokenSource.Cancel (System.Boolean throwOnFirstException) [0x00000] in <efe941bb62534dc3a62ceb1a818964a0>:0 
  at System.Threading.CancellationTokenSource.Cancel () [0x00000] in <efe941bb62534dc3a62ceb1a818964a0>:0 
  at GitCommands.AsyncLoader.Cancel () [0x00008] in <3eb8e9d083d04ef3adf987a8a1b67c27>:0 
  at GitUI.CommandsDialogs.FormCommit.Dispose (System.Boolean disposing) [0x00003] in <4ca0a9d3b3f3459fb67691431a63cda3>:0 
  at System.ComponentModel.Component.Dispose () [0x00000] in <1af2426e91d1474d92e5d471c4ca8f95>:0 
  at GitUI.GitUICommands+<>c__DisplayClass332_0.<StartCommitDialog>b__0 () [0x00034] in <4ca0a9d3b3f3459fb67691431a63cda3>:0 
  at GitUI.GitUICommands.DoActionOnRepo (System.Windows.Forms.IWin32Window owner, System.Boolean requiresValidWorkingDir, System.Boolean changesRepo, GitUIPluginInterfaces.GitUIEventHandler preEvent, GitUIPluginInterfaces.GitUIPostActionEventHandler postEvent, System.Func`1[TResult] action) [0x0002d] in <4ca0a9d3b3f3459fb67691431a63cda3>:0 
  at GitUI.GitUICommands.StartCommitDialog (System.Windows.Forms.IWin32Window owner, System.Boolean showOnlyWhenChanges) [0x00028] in <4ca0a9d3b3f3459fb67691431a63cda3>:0 
  at GitUI.CommandsDialogs.FormCommit.commitSubmoduleChanges_Click (System.Object sender, System.EventArgs e) [0x00025] in <4ca0a9d3b3f3459fb67691431a63cda3>:0 
  at System.Windows.Forms.ToolStripItem.OnClick (System.EventArgs e) [0x00019] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ToolStripMenuItem.OnClick (System.EventArgs e) [0x00090] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ToolStripMenuItem.HandleClick (System.Int32 mouse_clicks, System.EventArgs e) [0x00000] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ToolStripItem.FireEvent (System.EventArgs e, System.Windows.Forms.ToolStripItemEventType met) [0x00054] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.ToolStripItem.FireEvent(System.EventArgs,System.Windows.Forms.ToolStripItemEventType)
  at System.Windows.Forms.ToolStrip.OnMouseUp (System.Windows.Forms.MouseEventArgs mea) [0x00048] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ToolStripDropDown.OnMouseUp (System.Windows.Forms.MouseEventArgs mea) [0x00000] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control.WmLButtonUp (System.Windows.Forms.Message& m) [0x00078] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control.WndProc (System.Windows.Forms.Message& m) [0x001b4] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ScrollableControl.WndProc (System.Windows.Forms.Message& m) [0x00000] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ToolStrip.WndProc (System.Windows.Forms.Message& m) [0x00000] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.ToolStripDropDown.WndProc (System.Windows.Forms.Message& m) [0x00017] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control+ControlWindowTarget.OnMessage (System.Windows.Forms.Message& m) [0x00000] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.Control+ControlNativeWindow.WndProc (System.Windows.Forms.Message& m) [0x0000b] in <4b6c441381804088ab0fff508a3fbabf>:0 
  at System.Windows.Forms.NativeWindow.WndProc (System.IntPtr hWnd, System.Windows.Forms.Msg msg, System.IntPtr wParam, System.IntPtr lParam) [0x00085] in <4b6c441381804088ab0fff508a3fbabf>:0 
```

**Comment by zpangwin** [comment]
> I can confirm the occasional crash on Linux (Manjaro). The issue comes and goes (had it in the past), possibly related to installed fonts and/or mono version.

I too have had an occassional issue on Mint or Fedora but only very very rarely so I never actually bothered to capture the error log. For the most part, v2.51 still works great for me. Although in my case, when I do see issues it is generally in the form of weird GUI bugs related to mono that affect things like copy/paste, filters not being easy to clear/reset, etc. I am pretty sure the gui things are mono-specific though as I have seen similiar copy/paste oddities in Keepass2 whenever I use it instead of Keepassxc.

This might not be helpful for Arch/Manjaro distros but I'll include it just in case it has some small value for comparison purposes.... I have some install scripts in [this repo](https://github.com/zpangwin/linux-setup-scripts) that I use on my own Mint and Fedora setups, currerntly organized by distro name then app to be installed (I have been thinking about converting to distro-agnostic scripts to simplify maintenance so possible structure might change in the future). The [Fedora 33 git-extensions script](https://github.com/zpangwin/linux-setup-scripts/blob/master/fedora-33/git-extensions/install-git-extensions-2.5.sh) would probably more similar than the Mint ones but obv. still very different (especially dnf calls for installing mono vs pacman/yay). The main potential difference I can think that might be relevant tho would be if for the mono install, whether you installed the `mono-complete` package (or whatever the pacman/AUR equivalent is) vs just installing `mono-core` / `mono-runtime`. And even then, this would probably only matter if `mono-complete` includes fonts that are not present in the other package.

**Comment by zpangwin** [comment]
Also, want to note for the Git Extension maintainers, that I no longer have access to anything with Mac OSX on it (when I originally submitted this ticket, I had been installing it for my own use on a Macbook I had been issued at a former job which I have since moved on from). That had been my first experience with a Mac and especially in the file manager department I found them personally unpleasant to use, so I will be trying hard to never have to use one again in the future.

That being the case, I will leave it up to you guys whether you you prefer to leave it open or to close it. But I have no objections personally to closing due to: a) me no longer being able to test on Macs and b) there already being a ticket for this one the mono side of things ([Mono ticket #18330](https://github.com/mono/mono/issues/18330))
